feat(ProjectUI): Release Filter based on attachment availability

akapti opened the pull request:
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
* Added the `select option` to filter the release based on availability of attachment, in `Attachment Usage` tab of `project portlet`.

> Which issue is this pull request belonging to and how is it solving it? 
* #1103 

> Did you add or update any new dependencies that are required for your change?
* No


### How To Test?
> How should these changes be tested by the reviewer?
* selecting the `with attachment` option should display all the releases with attachments and also linked projects.
* selecting the `without attachment` option should display only the releases without attachments.
* selecting the `--Release filter--` option should reset the filter.

Screenshot for reference:
![image](https://user-images.githubusercontent.com/50870174/108158024-102e1800-710a-11eb-9517-182888a572d7.png)


> Have you implemented any additional tests?
* No

### Checklist
Must:
- [x] All related issues are referenced in commit messages and in PR

Signed-off-by: Abdul Kapti <abdul.kapti@siemens-healthineers.com>

mcjaeger commented:
thanks fro contributing this, but would it be flexible with respect to placement? I am not sure, if that is specifically something for the name, but maybe better put next to "Expand all | Collapse all"?

akapti commented:
> thanks fro contributing this, but would it be flexible with respect to placement? I am not sure, if that is specifically something for the name, but maybe better put next to "Expand all | Collapse all"?

Sure Michael, will make the necessary change.

JaideepPalit commented on frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/attachmentUsages.jspf:
```diff
@@ -38,12 +38,18 @@
+                <select class="mx-2" id="releaseFilter">
+                    <option value="" class="textlabel stackedLabel"> -- Release Filter -- </option>
+                    <option value="1" class="textlabel stackedLabel">With attachments</option>
+                    <option value="0" class="textlabel stackedLabel">Without attachments</option>
```
Since `Release Filter`, `With attachments`, `Without attachments` are hardcoded values.
Kindly use language.properties for I18N

akapti replied:
done.

JaideepPalit commented on frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/attachmentUsages.jspf:
```diff
@@ -38,12 +38,18 @@
             <core_rt:if test="${projectList.size() > 1 or (projectList.size() == 1 and not empty projectList.get(0).linkedReleases)}">
                 (<a href="#" id="expandAll" class="text-primary"><liferay-ui:message key="expand.all" /> </a>|
                 <a href="#" id="collapseAll" class="text-primary"> <liferay-ui:message key="collapse.all" /></a>)
+                <select class="mx-2" id="releaseFilter">
```
Would it be possible to use bootstrap class `form-control` like it is used in other places like obligation etc.
So that dropdown looks similar in entire sw360

akapti replied:
incorporated the changes.

JaideepPalit reviewed (changes requested):
Kindly look into some minor review comments.
Tested The PR.
Seems to be some issue.
- Issue 1 - Filter doesn't work for linked project without any attachment or releases. It still appears for `with attachment selection`
![image](https://user-images.githubusercontent.com/56534710/108315482-7df94300-71e1-11eb-9e70-8adc7730318f.png)

- Issue 2 - 
Filter works for level1 nodes of the tree and not for child nodes. For example in screenshot `a9` - a release,  does not have attachments but still appears for `with attachment` filter.
![image](https://user-images.githubusercontent.com/56534710/108315781-f3fdaa00-71e1-11eb-8de5-0172840476d8.png)

akapti commented:
Incorporated the review comments.

> * Issue 1 - Filter doesn't work for linked project without any attachment or releases. It still appears for `with attachment selection`

fixed the issue, excluded the linked projects because as per requirements filter should be applied only for linked releases and not the linked projects.

> * Issue 2 -
>   Filter works for level1 nodes of the tree and not for child nodes. For example in screenshot `a9` - a release,  does not have 
attachments but still appears for `with attachment` filter.

not an issue anymore as linked projects are excluded.

JaideepPalit reviewed (approved):
Code looks fine to me. Tested successfully
